api/src/blocks.rs

```diff
@@ -1,83 +1,31 @@
 // Copyright (c) Aptos
 // SPDX-License-Identifier: Apache-2.0
 
-use crate::{
-    context::Context,
-    failpoint::fail_point,
-    metrics::metrics,
-    param::{AddressParam, LedgerVersionParam, MoveIdentifierParam, MoveStructTagParam},
-    version::Version,
-};
-use std::cmp::min;
-
-use aptos_api_types::{
-    AccountData, Address, AsConverter, Error, HashValue, LedgerInfo, MoveModuleBytecode, Response,
-    TransactionId, TransactionOnChainData,
-};
-use aptos_types::{
-    account_config::AccountResource,
-    account_state::AccountState,
-    event::{EventHandle, EventKey},
-};
-
-use crate::param::Param;
+use crate::{context::Context, failpoint::fail_point, metrics::metrics, param::LedgerVersionParam};
 use anyhow::Result;
-use aptos_types::{
-    access_path::AccessPath, state_store::state_key::StateKey, transaction::Transaction,
-};
-use move_deps::move_core_types::{
-    identifier::Identifier,
-    language_storage::{ResourceKey, StructTag},
-    move_resource::MoveStructType,
-    value::MoveValue,
-};
-use std::convert::TryInto;
+use aptos_api_types::{Error, LedgerInfo, Response, TransactionId};
 use warp::{filters::BoxedFilter, Filter, Rejection, Reply};
 
-pub type BlockParam = Param<u64>;
-
-// GET /blocks/<address>
-pub fn get_block(context: Context) -> BoxedFilter<(impl Reply,)> {
-    warp::path!("blocks" / BlockParam)
+// GET /blocks/<version>
+pub fn get_block_info(context: Context) -> BoxedFilter<(impl Reply,)> {
+    warp::path!("blocks" / LedgerVersionParam)
```

Comment:
nit: "ledger_version" in our context means you want to query things as if the entire ledger is frozen at "ledger_version". LedgerVersionParam is used as an optional query param in the API code. 

In your case you probably just want to do `path!("blocks" / Version)`


also "blocks/12345"  makes it look like you are asking for the 12345th block, which is not the case, right? (Although I don't have a better naming scheme in mind).